test/compile_errors.zig

```diff
@@ -1,6 +1,25 @@
 const tests = @import("tests.zig");
 
 pub fn addCases(cases: *tests.CompileErrorContext) void {
+    cases.add(
+        "Use of && in place of `and`",
+        \\export fn entry() void {
+        \\    if (true && false) return;
+        \\}
+    ,
+        ".tmp_source.zig:2:14: error: expected token ')', found '&&'",
+        ".tmp_source.zig:2:14: note: did you mean to use `and`?",
+    );
+
+    cases.add(
+        "Use of || in place of `or`",
+        \\export fn entry() void {
+        \\    if (1 || 0) return;
```

Comment:
Nitpick. This should probably use `true` and `false` instead of `0` and `1`.

Reply:
Used `0` and `1` so that we could check on any type, since we are running through `ir.cpp`, but we could also include boolean types to be thorough.

Reply:
memo: added another test for `true` and `false`